.github/dependabot.yml

```diff
@@ -0,0 +1,24 @@
+version: 2
```

Comment:
Can you explain what this file does? I don't see any mentions to dependabot. Wouldn't the latest version of cargo packages/github-action packages be fetched when running the other workflows?

Reply:
Sorry for the wall of text ;) 

Dependabot was initially a service provided as a GitHub app by a company but got bought up by GitHub in 2019 (https://dependabot.com/blog/hello-github/).

It's a service that keeps dependencies up-to-date and informs you about security vulnerabilities in crates and recommends a version that has those fixed. It's sadly not enabled by default but has to be enabled inside the repository's settings under the `Security` tab.

AFAIK, you are correct. It will automatically download the newest version of either cargo or GitHub Actions (GA). GAs are usually fixed on their major version and cargo's resolution algorithm follows semver and will use the latest matching version (meaning that it bumps patch to the highest possible number or it bumps minor to the highest possible number if the minor has not been fixed by the maintainers). 

Back to this crate, I added the file as the dependencies for crates are denoted with their patch version. Doesn't make a difference when building but having up-to-date version strings inside of Cargo.toml seemed worth it to me considering that it literally is only a single button press on `Merge` on almost all of the PRs dependabot will open.
It would also come in handy in rare cases where a patch/minor version actually introduces a breaking change. Dependabot would open a PR with an according version bump and CI would fail if the public API changed. This either means a maintainer has to make the version requirement stricter by forcing a version below the newest one to prevent breaking builds from users or would have to update the codebase and use the newest version.

I do see why someone might not want this kind of service though.

Reply:
Oh I'm silly, I thought this was a workflow file and I see now that you're defining this as a config for Dependabot. Okay. Do you know if this has any advantages compared to adding the Dependabot integration to a project?

Reply:
I think both are still possible. The integration (GitHub app) is the "legacy" way to do it and using the configuration file uses the GitHub-native bot: https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/

They are basically the same. GitHub's blog post contains a section on moving to the GitHub-native implementation, i.e. creating a configuration file. I'd say that we should probably stick to that. It also shows how to enable it in the settings so that security updates are also being done ;)

Reply:
SGTM